### ISSUE TITLE
Whitespace issue when using else if

### ISSUE THREAD

**Issue by wolfjazz** [wycats/handlebars.js#1716]
Hi,

When I run the following example (please note the indentation of the inner block and that an else-if is used)
```
{{#if test}}
    {{#if test.a}}
    hihi
    {{else if test.b}}
    hoho
    {{else}}
    huhu
    {{/if}}
    hehe
{{/if}}
```

the result I get for `{ test: { c: true, }, }` is that hehe is intended twice:
```
    huhu
        hehe
```

<br>

However, when I try the same example with a nested if-else block instead the else if,
```
{{#if test}}
    {{#if test.a}}
    hihi
    {{else}}
    {{#if test.b}}
    hoho
    {{else}}
    huhu
    {{/if}}
    {{/if}}
    hehe
{{/if}}
```

I get the result I would expect for both cases, namely that both words are indented just once:
```
    huhu
    hehe
```

<br>

You can test that for yourself in the [handlebars playground](https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%23if%20test%7D%7D%5Cn%20%20%20%20%7B%7B%23if%20test.a%7D%7D%5Cn%20%20%20%20hihi%5Cn%20%20%20%20%7B%7Belse%20if%20test.b%7D%7D%5Cn%20%20%20%20hoho%5Cn%20%20%20%20%7B%7Belse%7D%7D%5Cn%20%20%20%20huhu%5Cn%20%20%20%20%7B%7B%2Fif%7D%7D%5Cn%20%20%20%20hehe%5Cn%7B%7B%2Fif%7D%7D%22%2C%22partials%22%3A%5B%5D%2C%22input%22%3A%22%7B%20test%3A%20%7B%20c%3A%20true%2C%20%7D%2C%20%7D%5Cn%22%2C%22output%22%3A%22%20%20%20%20huhu%5Cn%20%20%20%20%20%20%20%20hehe%5Cn%22%2C%22preparationScript%22%3A%22Handlebars.registerHelper('loud'%2C%20function%20(aString)%20%7B%5Cn%20%20%20%20return%20aString.toUpperCase()%5Cn%7D)%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.6%22%7D).

So I think I've found a Handlebars bug here. Or am I just doing something wrong?

**Comment by Akazm** [comment]
I'm facing the same issue in a different context.

**Comment by mickprev** [comment]
I am also facing this problem in my project. Does anyone have any idea on how to fix this bug?

**Comment by Mosmaz07** [comment]
> I am also facing this problem in my project. Does anyone have any idea on how to fix this bug?

The only solution that worked for me was not using else if at all but using nested if‘s as can be seen in my second example above.

**Comment by mickprev** [comment]
> > I am also facing this problem in my project. Does anyone have any idea on how to fix this bug?
> 
> The only solution that worked for me was not using else if at all but using nested if‘s as can be seen in my second example above.

Yes it's a shame to have to use this syntax to fix the bug. I would like to understand where the problem comes from :s

**Comment by theodorejb** [comment]
The reason for this bug is that when a block has chained `{{else if}}` branches, the parser's whitespace control code only strips a trailing indent from the first branch's body, rather than treating all branches consistently. I opened handlebars-lang/handlebars-parser#31 to fix this.
